docs/api_extra.rst

```diff
@@ -325,10 +325,35 @@ include directive:
          uses :ref:`intrusive reference counting <intrusive>`, the added
          indirection and ownership tracking removes the need for extra copies.
 
-     You should **never** use this class to bind pointer-valued vectors
-     ``std::vector<T*>`` when ``T`` does not use intrusive reference counting.
-     Some kind of ownership tracking (points 2 and 3 of the above list) is
-     needed in this case.
+         (It is usually unsafe to use this class to bind pointer-valued
+         vectors ``std::vector<T*>`` when ``T`` does not use intrusive
+         reference counting, because then there is nothing to prevent the Python
+         objects returned by ``__getitem__`` from outliving the C++ ``T``
+         objects that they point to. But if you are able to guarantee through
+         other means that the ``T`` objects will live long enough, the intrusive
+         reference counting is not strictly required.)
+
+   .. note::
+
+      Previous versions of nanobind (before 2.0) and pybind11 would instead
+      return Python objects from ``__getitem__`` that wrapped *references*,
+      meaning that they were only safe to use until the next insertion or
+      deletion in the vector they were drawn from. As discussed above, any
+      use after that point could **corrupt memory or crash your program**,
+      which is why reference semantics are no longer the default.
+
+      If you truly need the unsafe reference semantics, and you
+      promise that everyone who uses your bindings will be mindful
+      of the memory layout and reference-invalidation rules of the
+      underlying C++ container type, you can request the old behavior
+      by passing a second template argument of
+      :cpp:enumerator:`rv_policy::reference_internal` to
+      :cpp:func:`bind_vector`. This will override nanobind's usual
+      choice of :cpp:enumerator:`rv_policy::copy` for ``__getitem__``.
+
+      .. code-block:: cpp
+
+         nb::bind_vector<std::vector<MyType>, nb::rv_policy::reference_internal>();
```

Comment:
Can you give an example of something that might crash the Python session? (e.g appending an element). Just to show to people how bad it is.